### ISSUE TITLE
Provide a progress indicator on windows installation

### ISSUE THREAD

**Issue by djallado** [learningequality/installers#385]
## Summary

*User may close the terminal during installation and may cause the `KA Lite` become problematic. It's good to put a progress indicator during installation of `KA Lite`  and don't show the terminal window anymore.*

## Branch or installer

*Did issue occur on the current `develop` branch or on installation source (please specify OS and version)?*

 - Installer: Windows
 - Version: 0.16.x

or:

 - Branch: develop

## Screenshots

![terminal](https://cloud.githubusercontent.com/assets/8664071/14137560/d5dde598-f69a-11e5-9053-38ac47e30ac0.png)
Terminal window that popup during installation. User may possibly close this and it will cause the `KA Lite` become problematic.

![finish](https://cloud.githubusercontent.com/assets/8664071/14137566/dc0a8e58-f69a-11e5-92d5-9e9617f370a2.png)
Completing the `KA Lite` setup wizard. User may finish the installation even if he/she close the terminal window. 

## How to reproduce

1. Install 0.16.x release of `KA Lite` [from dungeon] (http://dungeon.learningequality.org/browse/KL-KALITE016-208/artifact/shared/Windows-installer/KALiteSetup-0.16.0.exe).
1. Follow the `KA Lite Setup Wizard`.
1. Terminal windows may popup, just wait it until it disappear. 
1. Click on `Finish` button in `KA Lite Setup Wizard`.

**Comment by ruimalheiro** [comment]
Hi @djallado ,

The thing here is that the installer doesn't check for the return codes of each program it calls, which basically means that the installer doesn't have control over the external calls and it proceeds assuming everything went OK.

I think the right solution here would be to give the installer control and hide the cmd window and instead show a progress indicator (as you suggest), as part of the installer or create new window for that purpose.

**Comment by MCGallaspy** [comment]
So I've tried two different approaches with the windows installer, and I feel both have significant drawbacks:

1. Hide the cmd prompt so the user can't inadvertently cancel it. But then some of these external calls are very long running, and the installer wizard gives no indication of progress. It gives the user the impression that something is wrong.
2. Showing the cmd prompt creates the possibility of the user closing it, and then installation proceeding although a necessary step has terminated early. @jamalex additionally contends that the cmd prompt is inherently distressing to users.

However I still think 2 is preferable to 1, because it *shows* the user that something is happening. Though indeed some of the long-running mgmt commands we run *don't* produce very much output and as in 1 gives the impression that something is wrong.

A better solution is to create mgmt commands with nice output, meaning that they produce output often enough to give the impression that something is *happening* even for long-running processes and additionally the output should not be too cryptic so that users aren't intimidated by it. (Although I feel that we're not giving users enough credit if we say that *any* cmd prompt is bad.)

Additionally, I think it should be possible to terminate the installation if the user has closed a command prompt, either by inspecting return codes or by identifying some other condition that indicates successful completion. We should do that too.

**Comment by jamalex** [comment]
> A better solution is to create mgmt commands with nice output, meaning that they produce output often enough to give the impression that something is happening even for long-running processes

Agreed, and then the GUI could be updated to read that output as well, and show its own progress bar or "status", without needing to pop up new windows.

> Although I feel that we're not giving users enough credit if we say that any cmd prompt is bad.

It's also about continuity -- if the install started in a console, it makes sense to have progress shown there. If the install is happening within the context of a GUI, popping up a separate (very different) window breaks up the seamless install flow and could be confusing.

To re-emphasize what I said in the other issue, I don't think this is a blocker for anything, and maybe we won't get to it in the context of KA Lite at all. But thinking about it will be helpful as we build out Kolibri -- e.g. designing long-running mgmt commands so that they give frequent updates and show progress.

**Comment by rtibbles** [comment]
:+1:

**Comment by benjaoming** [comment]
@mrpau-richard is this still relevant?
